### PR TITLE
planner/core: pass the elems info to fieldtype only for exchanger

### PR DESCRIPTION
**Pull request by hanfei1991**
### What problem does this PR solve?
Problem Summary:
TiFlash must know the elems info of enum type when decoding the enum data. Especially in MPP mode, we can't get the table schema after shuffle, then we must pass the elems to the tipb plans.

### Related changes

- PR to update `pingcap/tipb`: https://github.com/pingcap/tipb/pull/217

### Check List 

Tests 

- Unit test

### Release note 

- planner/core: pass the elems info to fieldtype only for exchanger

### CODE REVIEW

File: executor/tiflash_test.go [hunk 1]
```diff
@@ -271,3 +271,20 @@ func (s *tiflashTestSuite) TestPartitionTable(c *C) {
...
+	tk.MustExec("alter table t set tiflash replica 1")
+	tb := testGetTableByName(c, tk.Se, "test", "t")
+	err := domain.GetDomain(tk.Se).DDL().UpdateTableReplicaInfo(tk.Se, tb.Meta().ID, true)
+	c.Assert(err, IsNil)
+	tk.MustExec("insert into t values(1,'aca')")
+	tk.MustExec("insert into t values(2,'bca')")
+	tk.MustExec("insert into t values(3,'zca')")
+	tk.MustExec("set @@session.tidb_isolation_read_engines=\"tiflash\"")
+	tk.MustExec("set @@session.tidb_allow_mpp=ON")
+	tk.MustQuery("select t1.b from t t1 join t t2 on t1.a = t2.a").Check(testkit.Rows("aca", "bca", "zca"))
```
Comment (fzhedu) [inline executor/tiflash_test.go:289]: add a sort to guarantee the output order?

Comment (hanfei1991) [reply executor/tiflash_test.go]: done

**Review by fzhedu — COMMENTED** [review]
/LGTM

File: planner/core/plan_to_pb.go [hunk 2]
```diff
@@ -280,7 +281,13 @@ func (e *PhysicalExchangeReceiver) ToPB(ctx sessionctx.Context, storeType kv.Sto
 
 	fieldTypes := make([]*tipb.FieldType, 0, len(e.Schema().Columns))
 	for _, column := range e.Schema().Columns {
-		fieldTypes = append(fieldTypes, expression.ToPBFieldType(column.RetType))
+		pbType := expression.ToPBFieldType(column.RetType)
+		if column.RetType.Tp == mysql.TypeEnum {
+			for _, e := range column.RetType.Elems {
+				pbType.Elems = append(pbType.Elems, e)
+			}
```
Comment (lzmhhh123) [inline planner/core/plan_to_pb.go:288]:
```suggestion
			pbType.Elems = append(pbType.Elems, column.RetType.Elems...)
```

File: store/mockstore/unistore/cophandler/mpp.go [hunk 3]
```diff
@@ -132,7 +132,13 @@ func (b *mppExecBuilder) buildMPPExchangeReceiver(pb *tipb.ExchangeReceiver) (*e
 	}
 
 	for _, pbType := range pb.FieldTypes {
-		e.fieldTypes = append(e.fieldTypes, expression.FieldTypeFromPB(pbType))
+		tp := expression.FieldTypeFromPB(pbType)
+		if tp.Tp == mysql.TypeEnum {
+			for _, elem := range pbType.Elems {
+				tp.Elems = append(tp.Elems, elem)
+			}
```
Comment (lzmhhh123) [inline store/mockstore/unistore/cophandler/mpp.go:139]: ditto.

**Review by lzmhhh123 — APPROVED** [review]

**Pull request merged**